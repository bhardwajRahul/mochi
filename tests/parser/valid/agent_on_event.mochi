stream Temp {
  value: float
}

agent Logger {
  var last: float = 0.0
  on Temp as t {
    last = t.value
  }
  intent lastTemp(): float {
    return last
  }
}
