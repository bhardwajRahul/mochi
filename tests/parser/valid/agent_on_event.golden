(program
  (stream Temp (field "value:float"))
  (agent Logger
    (on Temp
      (assign last
        (selector value (selector t))
      )
    )
    (intent lastTemp
      (type float)
      (return (selector last))
    )
  )
)

